src/TC_util.cpp

```diff
@@ -0,0 +1,58 @@
+#include "TC_util.h"
+
+#include <string.h>
+
+#include "Devices/Serial_TC.h"
+
+// Example: strscpy(buffer, source, sizeof(buffer));
+int strscpy(char *destination, const char *source, unsigned long sizeOfDestination) {
+  unsigned long sourceLength = strnlen(source, sizeOfDestination);
+  if (sourceLength < sizeOfDestination) {
+    // Put a null terminator in the final byte
+    *((char *)memcpy(destination, source, sourceLength) + sourceLength) = '\0';
+    return 0;
+  } else {
+    // Put a null terminator in the final byte
+    *((char *)memcpy(destination, source, sizeOfDestination - 1) + sizeOfDestination - 1) = '\0';
+    serial(F("WARNING! String \"%s\" was truncated to \"%s\""), source, destination);
+    // TODO: Log a WARNING that a string was truncated
+    return 1;
+  }
+}
+
+// In this function, the source is treated as an address in PROGMEM, not RAM
+// Example: strscpy(buffer, source, sizeof(buffer));
```

Comment:
Is the example supposed to be showing how to use this function? If so, should it have `_P` on it? It feels like this was just copied from line 7. Is it usually used with something like `(PGM_P)F("...")` as the source?

Reply:
Yes, this was copied and not edited, and your suggested fix is good.